expfmt/text_parse_test.go

```diff
@@ -384,6 +385,73 @@ request_duration_microseconds_count 2693
 				},
 			},
 		},
+
+		// 5: Help string present but empty will result in empty help for metric family
+		{
+			in: `
+# HELP empty_help_string
+# TYPE empty_help_string counter
+empty_help_string 0
+`,
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("empty_help_string"),
+					Help: proto.String(""),
+					Type: dto.MetricType_COUNTER.Enum(),
+					Metric: []*dto.Metric{
+						{
+							Counter: &dto.Counter{
+								Value: proto.Float64(0),
+							},
+						},
+					},
+				},
+			},
+		},
+		// 6: Omitting help string produces nil help for metric family
+		{
+			in: `
+# TYPE no_help_string counter
+no_help_string 0
+`,
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("no_help_string"),
+					Help: nil,
+					Type: dto.MetricType_COUNTER.Enum(),
+					Metric: []*dto.Metric{
+						{
+							Counter: &dto.Counter{
+								Value: proto.Float64(0),
+							},
+						},
+					},
+				},
+			},
+		},
+		// 7: Two Help strings present but empty will overwritten
+		{
+			in: `
+# HELP should_not_be_empty_help_string
+# HELP should_not_be_empty_help_string not empty
+# TYPE should_not_be_empty_help_string counter
+should_not_be_empty_help_string 0
```

Comment:
As discussed in the other thread, I think this should not be allowed.

(As another side thought: If it were allowed, it would raise the question what happens if the non-empty HELP string comes first and _then_ you get an empty one. I think it would error out with the current code. And then we are in the situation where order suddenly decides between "totally OK" and "absolutely invalid". Again, I would prefer to just cleanly error out in either case.)

My whole point would be invalid if there _were_ a valid use case to have two HELP entries (of which the first is empty). But I doubt there is one…